xlators/protocol/server/src/server-rpc-fops_v2.c

```diff
@@ -3186,19 +3215,44 @@ int
 server4_lookup_resume(call_frame_t *frame, xlator_t *bound_xl)
 {
     server_state_t *state = NULL;
+    dict_t *xdata = NULL;
 
     state = CALL_STATE(frame);
 
     if (state->resolve.op_ret != 0)
         goto err;
 
-    if (!state->loc.inode)
+    xdata = state->xdata ? dict_ref(state->xdata) : dict_new();
+    if (!state->loc.inode) {
         state->loc.inode = server_inode_new(state->itable, state->loc.gfid);
-    else
+        if (xdata) {
+            int ret = dict_set_int32(xdata, GF_NAMESPACE_KEY, 1);
+            if (ret) {
+                gf_msg_debug(
+                    THIS->name, 0,
+                    "dict set (namespace) failed (path: %s), continuing",
+                    state->loc.path);
+            }
+            if (state->loc.path && (state->loc.path[0] == '<')) {
+                /* This is a lookup on gfid : get full-path */
+                ret = dict_set_int32(xdata, "get-full-path", 1);
+                if (ret) {
+                    gf_msg_debug(
```

Comment:
Can we continue in case of failure ?

Reply:
As I mentioned, its for information right now, and made the log as INFO for now. Working on a recursive lookup for handling gfid based lookups. Should be finished before next major release, so I believe this is good to get in.